### PR TITLE
Sign in for regular users

### PR DESCRIPTION
**Pull request by vctr90**
Hi there,

This is just a pr to add user authentication/authorization flows with JWT. It responds to #10 .

Feedback will be appreciated.

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  EXPIRATION_TIME_FOR_WEB_TOKEN: '2h'
```
Comment (rkmax) [inline config.js]: I think would be nice if you move this to some environment value and load here on the config.js every environment value that you want, in that way you will have all the usage of all the environmental values in just one place and also you may have default values

```js
module.exports = {
  EXPIRATION_TIME_FOR_WEB_TOKEN: process.env.EXPIRATION_TIME_FOR_WEB_TOKEN || '2h',
  SECRET: process.env.SECRET || 'the default secret',
  SALT_DEFAULT_WORK_FACTOR: JSON.parse(process.env.SALT_DEFAULT_WORK_FACTOR || '10')
  // ...
};
```

File: docker-compose.yml [hunk 2]
```diff
@@ -40,6 +40,7 @@ services:
     working_dir: /home/node/app
     ports:
       - ${PORT}:${PORT}
+      - ${DEBBUG_NODE_PORT}:${DEBBUG_NODE_PORT}
```
[the remark this answers is not in the document]
Comment (vctr90) [reply docker-compose.yml]: Yes, you are right, thanks!

File: models/user.js [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
-const { addPasswordEncryptionPreSaveHook } = require('../db/schemas/user/utils.js');
+const _ = require('lodash');
+const {
+  addPasswordEncryptionPreSaveHook,
+  comparePassword
+} = require('../db/schemas/user/utils.js');
 
 addPasswordEncryptionPreSaveHook({ schema: userSchema, fieldToHash: 'password' });
-const DbUser = mongoose.model('Users', userSchema);
+const DbUser = (mongoose && mongoose.models && mongoose.models.Users) || mongoose.model('Users', userSchema);
```
[the remark this answers is not in the document]
Comment (vctr90) [reply models/user.js]: Totally worth it 100%, thanks.

File: config.js (hunk 1 above)
Comment (vctr90) [reply config.js]: Done!

File: docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (vctr90) [reply docker-compose.yml]: Done

File: models/user.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (vctr90) [reply models/user.js]: Done

**Comment by vctr90** [comment]
Merged guys, thanks so much for your reviews. It totally worth it!

**Pull request merged**